src/solver/rosenbrock.cu

```diff
@@ -61,5 +140,51 @@ namespace micm
       return kernel_duration;
     }
 
+    double NormalizedErrorDriver(double* d_y_old, double* d_y_new, 
+                                 double* d_errors, const size_t num_elements,
```

Comment:
For consistency with `AlphaMinusJacobianDriver()`, should you pass the `CudaMatrixParam` references to this driver function, instead of the raw pointers?

Reply:
Thanks Matt. Based on my understanding, once the `CudaVectorMatrix` and `CudaSparseVectorMatrix` class is introduced into MICM, we should not need the `CudaMatrixParam` and `CudaSparseMatrixParam` struct any more (I will revise all those functions including `AlphaMinusJacobianDriver()` later). I do not mind passing raw pointers to the CUDA kernels. We could surely group them into a struct and make the interface clean. But I think we then need to do `cudamalloc` again for that struct (or its members) and that is something we don't want (we want all the `cudamalloc` happens in the constructors if possible). Let me know if I have any misunderstandings here.

Reply:
Thanks @mattldawson for your comments. I have updated the function/kernel interface as suggested.